### PR TITLE
LCAM-1247

### PR DESCRIPTION
**Pull request by Mhowell494**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-1247)

- Refactor `SysGenDateDeserializer` to handle dates in ISO-8601 format.
- Remove test scenarios duplicated between controller and integration tests.

**Note:** There are also accompanying changes required to MAAT, which you can find [here](https://github.com/ministryofjustice/laa-maat-application/pull/505).

### CODE REVIEW

File: maat-orchestration/src/test/java/uk/gov/justice/laa/crime/orchestration/jackson/LocalDateTimeDeserializerTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class LocalDateTimeDeserializerTest {
+
+    private final JsonFactory factory = new JsonFactory();
+    private final ObjectMapper mapper = new ObjectMapper();
+    private final LocalDateTimeDeserializer deserializer = new LocalDateTimeDeserializer();
+
+    @Test
+    void givenValidDate_whenDeserializeIsInvoked_thenLocalDateTimeIsDeserialized() throws IOException {
+        JsonParser parser = factory.createParser("1633027200000");
```
Comment (venkat980) [inline maat-orchestration/src/test/java/uk/gov/justice/laa/crime/orchestration/jackson/LocalDateTimeDeserializerTest.java:24]: should we define a variable to hold 1633027200000 ?

File: maat-orchestration/src/test/java/uk/gov/justice/laa/crime/orchestration/jackson/LocalDateTimeDeserializerTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.laa.crime.orchestration.jackson;
```
Comment (venkat980) [inline maat-orchestration/src/test/java/uk/gov/justice/laa/crime/orchestration/jackson/LocalDateTimeDeserializerTest.java:1]: Any reason for chosing jackson as the package name, how about "serializer" (similar to mapper)?

**Review by venkat980 — APPROVED** [review]
Looks good. Just couple of trivial observations.

File: maat-orchestration/src/test/java/uk/gov/justice/laa/crime/orchestration/jackson/LocalDateTimeDeserializerTest.java (hunk 2 above)
Comment (Mhowell494) [reply maat-orchestration/src/test/java/uk/gov/justice/laa/crime/orchestration/jackson/LocalDateTimeDeserializerTest.java:1]: I did consider that, but they're actually custom deserializers, so I thought jackson would be a good name for all classes related to handling JSON.

File: maat-orchestration/src/test/java/uk/gov/justice/laa/crime/orchestration/jackson/LocalDateTimeDeserializerTest.java (hunk 1 above)
Comment (Mhowell494) [reply maat-orchestration/src/test/java/uk/gov/justice/laa/crime/orchestration/jackson/LocalDateTimeDeserializerTest.java:24]: Good point, I've pushed a new commit which addresses this.

**Review by venkat980 — APPROVED** [review]

**Pull request merged**